src/pages/api/juicebox/projectLogo/[projectId].page.ts

```diff
@@ -0,0 +1,93 @@
+import { createServerSupabaseClient } from '@supabase/auth-helpers-nextjs'
+import axios from 'axios'
+import { getLogger } from 'lib/logger'
+import { NextApiRequest, NextApiResponse } from 'next'
+import { Database } from 'types/database.types'
+import {
+  cidFromUrl,
+  ipfsGatewayUrl,
+  ipfsUriToGatewayUrl,
+  isIpfsUri,
+} from 'utils/ipfs'
+import { getProjectMetadata } from 'utils/server/metadata'
+
+const logger = getLogger('api/juicebox/projectLogo/[projectId]')
+
+async function findLogoFromDb(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  projectId: string,
+) {
+  const supabase = createServerSupabaseClient<Database>({ req, res })
+
+  const query = await supabase
+    .from('projects')
+    .select('logo_uri')
+    .eq('project_id', projectId)
+
+  if (query.error) throw query.error
+
+  return query.data?.[0]?.logo_uri as string | undefined
+}
+
+async function findLogoOnChain(projectId: string) {
+  const { logoUri } = (await getProjectMetadata(projectId as string)) ?? {}
+  return logoUri
+}
+
+/**
+ *
+ * @dev 2 network requests happen here:
+ * 1. fetch the project metadata CID (DB fetch is prioritized)
+ * 2. fetch the project logo image data
+ *
+ * Note, if DB request fails, we still try to fetch the logo from chain. This will be 2 more requests.
+ *
+ * Thus this endpoint should be cached.
+ * However, the ideal duration is TBD.
+ * Please consider changing (lowering) if it causes issues with users.
+ */
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  if (req.method !== 'GET') {
+    return res.status(404)
+  }
+
+  try {
+    const { projectId } = req.query
+    if (!projectId) {
+      return res.status(400).json({ error: 'projectId is required' })
+    }
+
+    // try getting logo from db. if for some reason its not there, try getting it from chain.
+    const logoUri =
+      (await findLogoFromDb(req, res, projectId as string)) ??
+      (await findLogoOnChain(projectId as string))
```

Comment:
if the project legit doesnt have a logo, then we are doing 2 sync GETS for nothing.

Should we just remove this line and solely rely on the DB?

Reply:
I think so 👍

Reply:
rad. will do